plugins/gateway/tyk/v3.2.2/plugin.go

```diff
@@ -107,10 +108,15 @@ func createTelemetry(res *http.Response, req *http.Request) (*models.Telemetry,
 	if err != nil {
 		return nil, fmt.Errorf("failed to read request body: %v", err)
 	}
+	// Return the content to the request body (since we read it)
+	req.Body = ioutil.NopCloser(bytes.NewBuffer(reqBody))
+
 	resBody, truncatedBodyRes, err := readBody(res.Body)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read response body: %v", err)
 	}
+	// Return the content to the request body (since we read it)
+	res.Body = ioutil.NopCloser(bytes.NewBuffer(resBody))
```

Comment:
instead of duplicating the logic is it not possible to make `readBody` restore the content? maybe a pointer to the `io.ReadCloser` will need to be sent to the function.

Reply:
1. body is an interface so not sure I can pass it as a pointer
2. For this to work we need to pass the request/response itself, hence open two functions instead of one
3. It is not always necessary to copy the body back, since in some cases (like the gopacket plugin) we dont need the body after we read it